packages/optimizer/lib/transformers/OptimizeImages.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const {hasAttribute, nextNode, firstChildByTag} = require('../NodeUtils');
+const {skipNodeAndChildren} = require('../HtmlDomHelper');
+const {isValidImageSrcURL} = require('../URLUtils');
+
+const MAX_IMG_SIZE = 820;
+const MIN_WIDTH_TO_ADD_SRCSET_IN_RESPONSIVE_LAYOUT = 300;
+const NUM_SRCSET_DPR = [1.0, 2.0, 3.0];
+const SRCSET_WIDTH = [
+  39,
+  47,
+  56,
+  68,
+  82,
+  100,
+  120,
+  150,
+  180,
+  220,
+  270,
+  330,
+  390,
+  470,
+  560,
+  680,
+  820,
+  1000,
+  1200,
+];
+
+/**
+ * Default implementation that does not perform any kind of image optimization, but generates a `srcset` string
+ * by appending the width to the file name (e.g. image.jpg => image.32w.jpg).
+ *
+ * @param {string} src - the image src
+ * @param {number} width - the required widths (in px)
+ * @returns {string|undefined} - the image URL or undefined if no image is available in this dimension
+
+ */
+const DEFAULT_IMAGE_OPTIMIZER = (src, width) => {
```

Comment:
is this private? theres a lot of assumptions below (src exists, and is a string, with exists, etc)

Reply:
will turn this into a sample, it's not useful this way